src/colmap/scene/reconstruction_io.cc

```diff
@@ -843,6 +1115,49 @@ void WriteCamerasBinary(const Reconstruction& reconstruction,
   WriteCamerasBinary(reconstruction, file);
 }
 
+void WriteFramesBinary(const Reconstruction& reconstruction,
+                       std::ostream& stream) {
+  THROW_CHECK(stream.good());
+
+  WriteBinaryLittleEndian<uint64_t>(&stream, reconstruction.NumFrames());
+
+  for (const frame_t frame_id : ExtractSortedIds(reconstruction.Frames())) {
+    const Frame& frame = reconstruction.Frame(frame_id);
+    WriteBinaryLittleEndian<frame_t>(&stream, frame_id);
+    WriteBinaryLittleEndian<rig_t>(&stream, frame.RigId());
+    WriteBinaryLittleEndian<uint8_t>(&stream, frame.HasPose() ? 1 : 0);
+    if (frame.HasPose()) {
```

Comment:
`WriteImages` requires the image to be registered (otherwise CamFromWorld throws), but this doesn't - isn't this inconsistent?

Reply:
Good callout. I changed it.